pkg/cli/cmd/resourceprovider/new/new.go

```diff
@@ -0,0 +1,161 @@
+/*
+Copyright 2023 The Radius Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package new
+
+import (
+	"context"
+	"errors"
+	"os"
+	"os/exec"
+
+	"github.com/radius-project/radius/pkg/cli"
+	"github.com/radius-project/radius/pkg/cli/clierrors"
+	"github.com/radius-project/radius/pkg/cli/cmd/commonflags"
+	"github.com/radius-project/radius/pkg/cli/framework"
+	"github.com/radius-project/radius/pkg/cli/output"
+	"github.com/spf13/cobra"
+)
+
+// NewCommand creates an instance of the `rad resourceprovider new` command and runner.
+func NewCommand(factory framework.Factory) (*cobra.Command, framework.Runner) {
+	runner := NewRunner(factory)
+
+	cmd := &cobra.Command{
+		Use:   "new",
+		Short: "Scaffold new resource provider",
+		Long: `Scaffold new resource provider
+
+This command outputs a JSON template that can be used to create a new resource provider.
+		
+Resource providers are the entities that implement resource types such as 'Applications.Core/containers'. Resource providers can be defined, registered, and unregistered by users.`,
+		Example: `
+# Scaffold a new resource provider called 'Applications.Example'
+rad resourceprovider new Applications.Example`,
```

Comment:
Why don't we just use `rad resourceprovider create`? Isn't it kind of confusing to have `new` and `create` for users?

Reply:
Create would be for a PUT operation against our API. We try to be consistent about the terms we use - since this isn't the same as other `create` commands, we're not calling it that.